cpp/src/io/parquet/page_enc.cu

```diff
@@ -433,8 +433,9 @@ __global__ void __launch_bounds__(128)
                                   max_RLE_page_size(col_g.num_rep_level_bits(), num_vals));
 
       if (num_rows >= ck_g.num_rows ||
-          (values_in_page > 0 && (page_size + fragment_data_size > this_max_page_size)) ||
-          rows_in_page + frag_g.num_rows > max_page_size_rows) {
+          (values_in_page > 0 && (page_size + fragment_data_size > this_max_page_size ||
+                                  rows_in_page + frag_g.num_rows > max_page_size_rows)) ||
+          rows_in_page >= max_page_size_rows) {
```

Comment:
When is this check relevant? As I understand this, if `rows_in_page >= max_page_size_rows`, then `values_in_page > 0` is also true and the `rows_in_page + frag_g.num_rows > max_page_size_rows` condition will be checked anyway.

Reply:
I'm not sure...put it back in just to be safe. I'll see if I can dream up some weird edge case where this is necessary :sweat_smile: